commit.go

```diff
@@ -319,7 +333,8 @@ func (b *Builder) Commit(ctx context.Context, dest types.ImageReference, options
 	}
 
 	var manifestBytes []byte
-	if manifestBytes, err = retryCopyImage(ctx, policyContext, maybeCachedDest, maybeCachedSrc, dest, "push", getCopyOptions(b.store, options.ReportWriter, nil, systemContext, "", false, options.SignBy), options.MaxRetries, options.RetryDelay); err != nil {
+	logrus.Debugf("Calling from Commit")
+	if manifestBytes, err = retryCopyImage(ctx, policyContext, maybeCachedDest, maybeCachedSrc, dest, "push", getCopyOptions(b.store, options.ReportWriter, nil, systemContext, "", false, options.SignBy, nil, nil, nil), options.MaxRetries, options.RetryDelay); err != nil {
```

Comment:
Rather then extending retryCopyImage, could you just pass in options...

Reply:
I think we're extending `getCopyOptions` here. Is this fine or are you thinking something like this?

```
options := getCopyOptions(...)
options.decryptionKey = ...
```

Reply:
I ready this wrong.  I was thinking the copy options and options.MaxRetries,  I hate functions which just keep growing their params.  Is there a way we can consolidate.

Reply:
Have a couple ideas ... But seems like this would be a separate PR.


For `getCopyOptions` it looks like the only special case is the `store`... Maybe something like this? I guess it would be a different PR for this though.

```
initCopyOptionsWithStore(store, &cp.Options{
                ReportWriter:          reportWriter,
                SourceCtx:             sourceCtx,
                DestinationCtx:        destinationCtx,
                ForceManifestMIMEType: manifestType,
                RemoveSignatures:      removeSignatures,
                SignBy:                addSigner,
                OciEncryptConfig:      ociEncryptConfig,
                OciDecryptConfig:      ociDecryptConfig,
                OciEncryptLayers:      ociEncryptLayers,
        })
```


Consolidating the retry options
```
type copyRetryOptions struct{
    maxRetries int
    retryDelay time.Duration
}
```